### ISSUE TITLE
FAR on a parameter should find all call sites passing value to that parameter

### ISSUE THREAD

**Issue by tmat** [dotnet/roslyn#59087]
It would be useful if FAR found not only the call sites that explicitly name the parameter, but also ones that pass value to the parameter but do not name it, especially when the parameter is optional. For optional parameters it is not easy to find all places that pass value to ("reference") the parameter.

**Version Used**: 
17.2.0 Preview 1.0 [32124.505.main]

**Steps to Reproduce**:

```C#
class C
{
    static void Main()
    {
        F();
        F(1);
        F(param: 1);
    }

    static void F(int param = 2)
    {
    }
}
```

**Expected Behavior**:

Finds 

```C#
F(1);
```

and 

```C#
F(param: 1);
```

**Actual Behavior**:
![image](https://user-images.githubusercontent.com/41759/151249705-fbb4a5a4-8616-404f-a5c2-4102336b1d82.png)

**Comment by CyrusNajmabadi** [comment]
@ryzngard Isn't this what value-tracking is for?  Does it work for this case?

**Comment by ryzngard** [comment]
Yup, although it looks like some things are getting duplicated for this case. Some of these are expected and likely just need some fit/finish to reduce in the UI, but others aren't. Here's a picture of running it on the example code 

![image](https://user-images.githubusercontent.com/475144/151297978-cb602cbb-7394-41b2-800e-7d035170250c.png)

**Comment by tmat** [comment]
So the point is to find call sites that pass the parameter and not display those that don't.

**Comment by tmat** [comment]
I think we can enhance FAR to find all call sites that pass an argument to the parameter (named or not) and tag results with "implicit" vs. "explicit". Then I can filter out what I'm not interested in.

**Comment by jinujoseph** [comment]
In Design review : To determine if Value tracking is sufficient for this ask

**Comment by genlu** [comment]
Design meeting conclusion:

ValueTracking would be able to show all cases but have a few drawbacks:
1. Less discoverable as a new feature
2. It's intended for a narrower use case therefore the UI might show more info than users would like if they just want all the callsites.
3. ValueTracking doesn't have way to differentiate implicit, explicit named and omitted reference of parameters at callsites. 
		
Therefore we'd want to support this in FAR, but with a new column in the FAR window for parameter ref kind, i.e. implicit, Explicit named, omitted (with default value used)

Bonus: Add an entry point for value tracking in FAR window.
